src/lib/db-types.ts

```diff
@@ -0,0 +1,123 @@
+// NOTE: see docs/firebase-schema.md to see a visual heirarchy of these interfaces
+
+export interface DBPortalUser {
+  version: "1.0";
+  self: {
```

Comment:
I know we talked about this at the meeting but `self` doesn't seem like the right name now that I see it in code. For one thing self is used heavily in MST and so it's a potential source of confusion, but more to the point it's really information that back-points to this object's parents. Perhaps `back` or `parent(s)` would be a better name?